examples/migration/recursive_fibonacci/fibonacci_single_task.h

```diff
@@ -0,0 +1,91 @@
+/*
+    Copyright (c) 2023 Intel Corporation
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+#ifndef SINGLE_TASK_HEADER
+#define SINGLE_TASK_HEADER
+
+#include "task_emulation_layer.h"
+
+#include <iostream>
+#include <numeric>
+#include <utility>
+
+extern int cutoff;
+
+long serial_fib_1(int n) {
+    return n < 2 ? n : serial_fib_1(n - 1) + serial_fib_1(n - 2);
+}
+
+struct single_fib_task : task_emulation::base_task {
+    enum class state {
+        compute,
+        sum
+    };
+
+    single_fib_task(int n, int* x) : n(n), x(x), s(state::compute)
+    {}
+
+    void execute() override {
+        switch (s) {
+            case state::compute : {
+                compute_impl();
+                break;
+            }
+            case state::sum : {
+                *x = x_l + x_r;
+                break;
+            }
+        }
+    }
+
+    void compute_impl() {
+        if (n < cutoff) {
+            *x = serial_fib_1(n);
+        }
+        else {
+            auto bypass = this->allocate_child_of_continuation_safe<single_fib_task>(n - 2, &x_r);
+            task_emulation::run_task(this->allocate_child_of_continuation_safe<single_fib_task>(n - 1, &x_l));
+
+            // Recycling
+            this->s = state::sum;
+            this->recycle_as_continuation();
```

Comment:
This is safe here because the bypass task is not spawned, and so it cannot complete before this->s is set. In general though, we are missing the recycle_as_safe_continuation function which would prevent a race in the case where both children were passed to run_task before compute_impl is done.  So recycling as continuation here assumes the developer guards against this race.

Reply:
Does `recycle_as_safe_continuation` assume that it should be called before any `spawn`?
Do you think we need it? It might be problematic because tasks doesn't have reference to itself only a children references.

Reply:
Not necessarily. recycle_as_safe_continuation makes *this a continuation and adds an implicit decrement of the ref_count after the task completes its initial execute (the one that does the recycle call). So it assumes that the reference count is N+1, where N is the number of children. Or at least that the reference count is incremented N+1 times (it might not actually reach N+1 if some children are spawned and complete). With continuation passing you don't wait inside the task body for the children to complete, so you just need to make sure that the reference counting works out and the continuation is not prematurely started (unless that's ok).  Since it may not be ok, the safe version is there to prevent it.